device/core/test/_sak_authentication_provider_test.js

```diff
@@ -0,0 +1,136 @@
+  });
+
+  describe('fromConnectionString', function () {
+    /*Tests_SRS_NODE_SAK_AUTH_PROVIDER_16_006: [The `fromConnectionString` method shall throw a `ReferenceError` if the `connectionString` parameter is falsy.]*/
+    [null, undefined, ''].forEach(function (badConnectionString) {
+      it('throws if the connection string is \'' + badConnectionString + '\'', function () {
+        assert.throws(function () {
+          SharedAccessKeyAuthenticationProvider.fromConnectionString(badConnectionString);
+        }, ReferenceError);
+      });
+    });
+
+    /*Tests_SRS_NODE_SAK_AUTH_PROVIDER_16_007: [The `fromConnectionString` method shall throw an `errors.ArgumentError` if the `connectionString` does not have a SharedAccessKey parameter.]*/
+    it('throws an ArgumentError if the connection string does not contain a shared access key', function () {
+      assert.throws(function () {
+        var fakeConnectionString = 'DeviceId=deviceId;HostName=host;SharedAccessKeyName=foo';
+        return SharedAccessKeyAuthenticationProvider.fromConnectionString(fakeConnectionString, 1, 1);
+      }, errors.ArgumentError);
+    });
+
+    /*Tests_SRS_NODE_SAK_AUTH_PROVIDER_16_008: [The `fromConnectionString` method shall extract the credentials from the `connectionString` argument and create a new `SharedAccessKeyAuthenticationProvider` that uses these credentials to generate security tokens.]*/
+    it('initializes the credentials from the connection string', function (testCallback) {
+      var fakeCredentials = {
+        deviceId: 'fakeDeviceId',
+        host: 'fake.host.name',
+        sharedAccessKey: 'fakeKey'
+      };
+      var fakeConnectionString = 'DeviceId=' + fakeCredentials.deviceId + ';HostName=' + fakeCredentials.host + ';SharedAccessKey=' + fakeCredentials.sharedAccessKey;
+
+      var sakAuthProvider = SharedAccessKeyAuthenticationProvider.fromConnectionString(fakeConnectionString, 1, 1);
+      sakAuthProvider.automaticRenewal = false;
+      sakAuthProvider.getDeviceCredentials(function (err, creds) {
+        assert.strictEqual(creds.deviceId, fakeCredentials.deviceId);
+        assert.strictEqual(creds.host, fakeCredentials.host);
+        assert.strictEqual(creds.sharedAccessKey, fakeCredentials.sharedAccessKey);
+        testCallback();
+      });
+    });
+  });
+
+  describe('updateSharedAccessSignature', function () {
+    /*Tests_SRS_NODE_SAK_AUTH_PROVIDER_16_004: [The `updateSharedAccessSignature` method shall save the `sharedAccessSignature` passed as a parameter and return it when `getDeviceCredentials` is called, until it gets renewed.]*/
+    it('updates the shared access signature', function (testCallback) {
+      var newSas = 'newSas';
+      var fakeCredentials = {
+        deviceId: 'fakeDeviceId',
+        host: 'fake.host.name',
+        sharedAccessKey: 'fakeKey'
+      };
+      var sakAuthProvider = new SharedAccessKeyAuthenticationProvider(fakeCredentials);
+      sakAuthProvider.automaticRenewal = false;
```

Comment:
is automaticRenewal part of the public contract?  If so, suggest adding requirements.  If not, suggest renaming -- at least add an underscore.

Reply:
removed

---
In reply to: [157025974](https://github.com/Azure/azure-iot-sdk-node/pull/179#discussion_r157025974) [](ancestors = 157025974)